### PR TITLE
Integrate GET Questions in Frontend

### PR DESCRIPTION
**Pull request by TejasNasre**
Closes: #784 

## Brief description of what is fixed or changed
We have get all questions in the the backend and also the questions has the upvote and downvote so i integrated in the frontend and added the UI for the question cards. 

## Types of changes
- [x] As soon as someone visits on Q&A page, there should be the [GET Questions](https://github.com/HITK-TECH-Community/Community-Website/blob/main/backend/app/routes/Q%26A/question/getAllQuestion.js) call.
- [x] The upvote functionality is integrated in frontend and downvote Also.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes does not break the current system and it passes all the current test cases.

## Screenshots

https://github.com/HITK-TECH-Community/Community-Website/assets/113225478/d098078e-4861-4694-9c62-b36b608f647b

questioncard

## Other Information
Please clear the test questions from mongodb database.

### CODE REVIEW

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
Please make the following changes - 
1. Add loader while the api call is happening, you can check other sections for reference
2. Show the toast message of success/failure of the api call

**Comment by TejasNasre** [comment]
Yeah I added the loader in the question section by doing conditional rendering but its not showing and the toast message will get on upvote or when the user gets on the q&a page please specify.

**Comment by Kajol-Kumari** [comment]
@TejasNasre 

For pt #1 please see the attached screenshot, i am not seeing the loader when get questions api call is happening. 
For pt #2 We should see toast for all the api calls, getQuestions, upvotes and downvotes as well, so that user remains informed of anything wrong happening during the api call

https://github.com/HITK-TECH-Community/Community-Website/assets/44888949/b2315155-b806-4426-9c69-54e9f4c24350

**Comment by TejasNasre** [comment]
Fix : Integrate-GET-Questions-in-Frontend

# Type Of Changes

- [x] loader when get questions api call is happening
- [x] toast for all the api calls, getQuestions, upvotes and downvotes as well
- [x] upvote and downvote both working

# Screen Video

https://github.com/HITK-TECH-Community/Community-Website/assets/113225478/197e0a32-efed-49f6-8cdf-113d1c1f1e50

File: frontend/src/pages/Q&A/Q&A.jsx
Comment (Kajol-Kumari) [inline frontend/src/pages/Q&A/Q&A.jsx:227]: we shouldn't be showing this toast message while fetching the data. Just loader animation while fetching is fine. Please check https://github.com/HITK-TECH-Community/Community-Website/blob/main/frontend/src/pages/ContactUs/ContactUs.jsx#L323-L328 this section for reference. For every api call, there can be success/failure and the toast should be showing success message in green if api is successful or failure message in red with error message. in your pr i don't see the failure scenario being handled. Please go through the code i shared here, it will give you the idea of toast implementation

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]

File: frontend/src/pages/Q&A/Q&A.jsx
Comment (TejasNasre) [reply frontend/src/pages/Q&A/Q&A.jsx:227]: ok i am working on this and removing the toast message on the loading.

Comment (TejasNasre) [reply frontend/src/pages/Q&A/Q&A.jsx]: @Kajol-Kumari Please Check Out Now.

**Comment by Kajol-Kumari** [comment]
@TejasNasre  let me know asap if you are willing to work on https://github.com/HITK-TECH-Community/Community-Website/issues/885 this issue?

**Comment by TejasNasre** [comment]
who is answering the questions.

**Comment by Kajol-Kumari** [comment]
> who is answering the questions.

It can be answered by anyone but will appear only after admin's approval. As far as i remember that was implemented, let me check this its been a while i have checked this codebase

**Comment by Kajol-Kumari** [comment]
seems like api does exist but need a little tweak to have a flag `isAdminApproved` so will keep this integration on hold till the time the api gets fixed :)

**Comment by TejasNasre** [comment]
I want your help - i recently worked on others repo and i raised some issue and they got merged but today in my forked repo i discard the commits unconditionally and the merge code is there on main but when I checked the PR all commits are deleted and some PR did not get merged yet and there commits also get  deleted so I will get points on GSSOC and what if I delete the fork repo and then then rebase it.

**Comment by Kajol-Kumari** [comment]
the points are only one the basis of labels assigned to the merged prs so yes, you will get the points for those prs

**Comment by TejasNasre** [comment]
What if I delete the fork repo for this project and then rebase code and I got merged one PR so I will get points?

**Comment by Kajol-Kumari** [comment]
all merged prs with correct label will give you required points

**Comment by TejasNasre** [comment]
Thanks @Kajol-Kumari For Helping

**Comment by Kajol-Kumari** [comment]
Hey @TejasNasre  i am assigning the levels based on the complexity of the issues only, if the required changes comes to be more than my expectation, i would be happy to revise the levels :)

**Comment by TejasNasre** [comment]
Ok Thank 👍🏻

**Pull request merged**
